storage/localstore/localstore.go

```diff
@@ -432,6 +437,7 @@ func New(path string, baseKey []byte, o *Options) (db *DB, err error) {
 func (db *DB) Close() (err error) {
 	close(db.close)
 	db.updateGCWG.Wait()
+	db.subscritionsWG.Wait()
```

Comment:
Not introduced by this PR only, but is it ok to do `Wait()` on an actual `Close` method, which is supposed to be for shutting down something? I mean, is it ensured that `Wait()` will not wait forever, e.g. because of blocked `go` routines?

Reply:
well if you want a clean shutdown you'd probably want to wait until all of the goroutines that use a certain resource exit no?
if they don't exit it means that there's a deadlock and that should be fixed

Reply:
As we talked, I've added a deadline of 5 seconds, that should be reasonable. And also I have added by @acud suggestion a goroutine profile dump, which we should improve int he future to control where it is written, currently it is in stdout.